frontend/js/index.js

```diff
@@ -329,7 +329,9 @@ function handleFailure(data) {
         displayError("The request timed out.")
     } else if (data.error == 'abort') {
         displayError("The request was aborted.")
-    } else if (isNaN(data.error)) {
+    } else if (typeof data.error === 'string') {
+        displayError(data.error);
+    } else if (isNaN(data.error + "")) {
```

Comment:
> Won't it print `true` as error code?
Okay yes i see, thats not suposed to be there. I have now changed it to `settings.error({error: 0});` which means that it will be an unkown error instead of printing true.

I think the error handling in general is not that great and could use some work...

Reply:
With `0` it will show `Unknown error occured, are you connecting to a Jellyfin Server?` because `isNaN(0 + '') -> false`.
I think we should show HTTP error code, if possible, so we can troubleshoot.

Reply:
> I think the real error code is set here:
[jellyfin-webos/frontend/js/ajax.js](https://github.com/jellyfin/jellyfin-webos/blob/ab6c23648b047b2d49ce0f5c6d4a42b1796f7ea6/frontend/js/ajax.js#L55)

this will  pass 200 since `event.target.status` is just the HTTP response code, but the issue happens on status 200.

I like your suggested change, since it makes more sense to check if the type is a number instead of checking if its not. The statement actually doesn't work since if i for example go to https://www.google.com and get a response of 404 the error displayed is:
<img width="767" alt="image" src="https://user-images.githubusercontent.com/66884216/175618375-5e18fd54-ec29-4cab-a17a-6867b7c419f6.png">
but with your suggested change it shows it correctly:
<img width="696" alt="image" src="https://user-images.githubusercontent.com/66884216/175618645-76e72175-56f4-4c24-a0ea-363741dfaa32.png">

I updated my fork with your change and fixed the mistake where i returned `true` instead of `0`

Reply:
> With `0` it will show `Unknown error occured, are you connecting to a Jellyfin Server?` because `isNaN(0 + '') -> false`. I think we should show HTTP error code, if possible, so we can troubleshoot.

I completely agree but the HTTP status code is 200 when the error happens. The error happens when it failed to parse the data. For example if it tries to parse HTML data, then it will fail because of a syntax error, then -> throw "The server did not return valid JSON data" the problem with returning the HTTP status code is that the status is 200 OK which is not useful for debugging. If the parsing failed and it wasn't a syntax error then throw an unknown error.

In my last commit to this fork, it will now correctly show the HTTP status code, see example above with google.com